File: src/main/java/com/nanaios/polygonal_tech/capability/interfaces/ICapability.java
package com.nanaios.polygonal_tech.capability.interfaces;

import com.nanaios.polygonal_tech.util.interfaces.IEventTarget;

/// 最も抽象化されたCapabilityインターフェース。
public interface ICapability extends IEventTarget {
    /// このCapabilityが入力可能かどうか
    boolean canInput();
    /// このCapabilityが出力可能かどうか
    boolean canOutput();
}

File: src/main/kotlin/com/nanaios/polygonal_tech/lib/capability/ICapability.kt
package com.nanaios.polygonal_tech.lib.capability

import com.nanaios.polygonal_tech.lib.util.sync.value.ISyncValue
import net.minecraft.nbt.CompoundTag
import net.minecraftforge.common.util.INBTSerializable

interface ICapability: ISyncValue, INBTSerializable<CompoundTag>